src/utils/validateIdentifierInputs.ts

```diff
@@ -0,0 +1,26 @@
+import { IdentifierInput } from "../types";
+import { fullPhoneNumberValidator } from "./validatePhoneNumbers";
+
+const isMatchRegex = (text: string, regex?: string): boolean => {
+  if (!regex) {
+    return true;
+  }
+  return new RegExp(regex).test(text);
+};
+
+export const validateIdentifierInputs = (
+  identifierInputs: IdentifierInput[]
+): boolean => {
+  for (const { value, validationRegex, textInputType } of identifierInputs) {
+    if (textInputType === "NUMBER" && isNaN(Number(value))) {
+      throw new Error("Invalid details");
+    }
+    if (!isMatchRegex(value, validationRegex)) {
+      throw new Error("Invalid details");
+    }
+    if (textInputType === "PHONE_NUMBER" && !fullPhoneNumberValidator(value)) {
+      throw new Error("Invalid contact number");
+    }
+  }
```

Comment:
Do you think there's a need to validate blank `value`s too?

Reply:
It's validated in a block before this in `checkoutCart`. Didn't add it here because there are other conditions sharing the same error message. Not sure which is better 🤔

Reply:
Oh i see, there's the check for `numUnverifiedTransactions`. Hmm, thinking that since this function can be used anywhere, should add the safeguards too, what do you think?

Reply:
Makes sense, will move that check here!

Reply:
Refactored :D